### PR TITLE
cli: clean up /build before wasp build

### LINKED ISSUE #210: Clean up /build dir before `wasp build`
This is to ensure that there are no left-overs from the previous `wasp build`.

### PR DESCRIPTION
**Pull request by kolharsam**
# Description

adds changes to clear the `build` folder while the `wasp build` command is being run.

Fixes #210 

## Type of change

Please select the option(s) that is more relevant.

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: waspc/cli/Command/Build.hs [hunk 1]
```diff
@@ -14,14 +14,28 @@ import CompileOptions (CompileOptions (..))
...
+    removeDirectoryRecursive,
+  )
 
 build :: Command ()
 build = do
   waspProjectDir <- findWaspProjectRootDirFromCwd
   let outDir =
         waspProjectDir </> Common.dotWaspDirInWaspProjectDir
           </> Common.buildDirInDotWaspDir
+      buildDirFilePath = toFilePath outDir
```
Comment (Martinsos) [inline waspc/cli/Command/Build.hs:29]: To keep naming consistent here in this function, I would rename buildDirFilePath to outDirFilePath. Or, rename outDir to buildDir, that is also ok! Maybe even better.

File: waspc/cli/Command/Build.hs [hunk 2]
```diff
@@ -14,14 +14,28 @@ import CompileOptions (CompileOptions (..))
...
+
+  liftIO $ putStrLn "Clearing the content of the .wasp/build directory..."
+  doesBuildDirExist <- liftIO $ doesDirectoryExist buildDirFilePath
+  if doesBuildDirExist
+    then liftIO $ do
+      removeDirectoryRecursive buildDirFilePath
```
Comment (Martinsos) [inline waspc/cli/Command/Build.hs:35]: I would move the "Clearing the content... " inside the if-then, and would remove "Nothing to clean ..." completely. So if there is no build dir, there is no messaging, and if there is one, we first say that we are clearing the content and then we say we sucessfully cleared it.

**Review by Martinsos — COMMENTED** [review]
Looking good! I suggested some minor changes, but otherwise it is all good -> once those are done we can merge :).

**Comment by kolharsam** [comment]
@Martinsos have updated the PR! :)

**Comment by Martinsos** [comment]
Great, I will merge this one!

Here are some ideas for the future issues:
- #191 -> this one should be relatively approachable / short.
- #207 -> this one is a bigger thing, it is about having a fancy way to show progress during `wasp start` execution. It requires some research and it is not a small task, but I think it could be pretty fun, and it is not all about Wasp internal but instead a mix of Wasp stuff and CLI visualization. I could help you with brainstorming this one and then splitting it into smaller issues that you could tackle one by one. Since this one is not urgent for now, it means you could work on it at your own tempo, as you wish. So, this makes sense if you would like to have an overarching issue that you can own and then work on it for some time in form of smaller issues.
- #8 -> This one is probably not a ton of code to get working, but it will require some research and testing, and I am actually not sure how much nor do I know the solution in advance.

**Comment by kolharsam** [comment]
Hey @Martinsos! Thanks for the details on all of the issues.
So here's what I will do:
1. I can take up #191 now. So yeah, I'd like it if there's a little bit more info on the issue regarding what needs to be done.
2. We can also start a thread for #207 on the issue so that I get know what I'm dealing with.
3. After these, if there's more info on #8, I can definitely take it up! :)

What does this sound like?

**Comment by Martinsos** [comment]
@kolharsam Ok, let's start with #191! I provided more information in the #191 regarding how to approach the issue -> I suggest we continue the discussion there. Once you write a message there, I will also be able to assign the issue to you.

#207 -> there is already quite some information there, I think that should be enough to start the discussion? As I mentioned, this issue is somewhat bigger and is more on the creative side, meaning it will require some research and brainstorming, but I think it could be really fun if you decide to commit to it - however it wouldn't make much sense to try to get it done quickly as it will certainly result in multiple smaller tasks.

#8 -> I could certainly try to provide some more info on this, but let's leave it for later for now then, until we get to it.

**Pull request merged**